Recipes/Recipes/View Controllers/MainViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  MainViewController.swift
+//  Recipes
+//
+//  Created by Jesse Ruiz on 9/30/19.
+//  Copyright © 2019 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class MainViewController: UIViewController {
+    
+    // MARK: - Properties
+    let networkClient = RecipesNetworkClient()
+    var allRecipes: [Recipe] = [] {
+        didSet {
+            filterRecipes()
+        }
+    }
+    var recipesTableViewController: RecipesTableViewController? {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    var filteredRecipes: [Recipe] = [] {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    
+    // MARK: - Outlets
+    @IBOutlet weak var textField: UITextField!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        networkClient.fetchRecipes { (recipes, error) in
+            if let anError = error {
+                NSLog("There was an error: \(anError)")
+            } else {
+                guard let recipe = recipes else { return }
+                self.allRecipes = recipe
```

Comment:
on line 41 remember you are setting the value of allRecipes inside of a network call.   With this in mind there is a didSet on allRecipes that then makes changes to the UI.  So you must make sure you perform this action on line 41 on the main thread 
``` 
DispatchQueue.main.async {
                self.allRecipes = recipe
                }
```

Reply:
This will get rid of the purple thread errors errors